R/ard_categorical.R

```diff
@@ -45,24 +45,27 @@
 #'
 #' In such cases, use the `denominator` argument to specify a new definition
 #' of `"N"`, and subsequently `"p"`.
-#' The argument expects a data frame, and the data frame must include the columns
-#' specified in `ard_categorical(by=)` (strata columns are not considered).
-#' The updated `N` and `length` elements will be updated to be calculated as
-#' the number of rows in each combination of the `by` variables.
+#' The argument expects the following inputs:
+#' - a data frame. Any columns in the data frame that overlap with the `by`/`strata`
+#'   columns will be used to calculate the new `"N"`.
+#' - an integer. This single integer will be used as the new `"N"`
+#' - a string: one of `c("column", "row", "cell")`. `"column"` is equivalent
+#'   to `denominator=NULL`. `"row"` gives 'row' percentages where `by`/`strata`
+#'   columns are are the 'top' of a cross table, and the variables are the rows.
+#'   `"cell"` gives percentages were the denominator is the number of non-missing
+#'   rows in the source data frame.
+#' - a structured data frame. The data frame will include columns from `by`/`strata`.
+#'   The last column must be named `"...ard_N..."`. The integers in this column will
+#'   be used as the updated `"N"` in the calculations.
 #'
 #' @return a data frame
-#' @name ard_categorical
 #' @export
 #'
 #' @examples
 #' ard_categorical(ADSL, by = "ARM", variables = "AGEGR1") |>
 #'   flatten_ard()
-NULL
-
-#' @rdname ard_categorical
-#' @export
 ard_categorical <- function(data, variables, by = NULL, strata = NULL,
-                            statistics = everything() ~ categorical_variable_summary_fns(),
+                            statistics = everything() ~ categorical_variable_summary_fns(c("n", "p", "N")),
```

Comment:
I was thinking of the tfrmt users here. I thought if we listed the stats here, it would be more obvious to them how to remove `N`? What do you think? I am fine with either!

Reply:
That makes perfect sense!
It would be good to keep consistency across the ard_* functions, and since the continuous/missing stats have a larger set of options, I think it's fine to leave it as `categorical_variable_summary_fns()`

Reply:
OK, updated!